### DISCUSSION TITLE
Developers guide on docs?

### DISCUSSION THREAD

**Discussion by Gonzalo-Mier** [Fields2Cover/Fields2Cover#12] [General]
Should we write a developers documentation to ease the learning curve of the library? Or should we prioritize other things?

**Comment by wilcobonestroo** [comment]
Hi Gonzalo, We are going to work with the library from September. I tried to install it according to the documentation and that went fine. I am not sure what issues we will run into when we actually start developing. We will let you know what things are hard to do. And if we know how to do things, we will also use PRs to update the documentation. Best regards, Wilco

**Reply by Gonzalo-Mier** [reply]
Hi Wilco. I will also add documentation before September to use the python interface. PR are more than welcome. Thank you for your time. Best regards, Gonzalo

**Comment by divya0056** [comment]
I  am not a programmer, but some knowledge of python and mostly Arduino. I have a large farm and trying to build a mower. I really appreciate your willingness to help the farmers. Could you please put and example to incorporate gps in your library. So far I have a tractor which can go to point a to point b using GPS but it's difficult for me to actually use this library. I guess the ultimate moto of your work is simply a farmers life

Regards

**Reply by Gonzalo-Mier** [reply]
Hello divya0056, I'll write more docs about that soon, when I have a bit more time.

For a quick help, for gps conversion to something in meters you need the [Transform class](https://github.com/Fields2Cover/Fields2Cover/blob/main/include/fields2cover/utils/transformation.h), and the [Field data type](https://github.com/Fields2Cover/Fields2Cover/blob/main/include/fields2cover/types/Field.h).

I guess you are using the python interface, so:

0. You already have the shapes  your field as a `f2c.Cells` called `cells` (for the example
1. Initialize a `f2c.Field` with `cells` and set your geolocalization system (GPS in this case):
```
field = f2c.Field();
field.field = cells;
field.setEPSGCoordSystem(4326);   # GPS == EPSG:4326
```

3. Transform field to a geolocalization system that works on meters on your place.
```
f2c.Transform_transform(field, "UTM:32U");     # Works with EPSG and UTM
```

5. Follow the other tutorials to create a `f2c.Path` called `path`
6. Transform path again to GPS:
```
f2c.Transform_transform(path, field, "EPSG:4326");
```

Bad news: step 6 is not working yet on python. I've just found this issue while writing the comment. I'll work soon on it, but for now you can start playing with the library. On C++, this example works fine.

Best regards :)

**Reply by divya0056** [reply]
Thank you, I will got it a try.

**Reply by Gonzalo-Mier** [reply]
You can do now the step 6 with the `improve-parser` branch like:
```
 f2c.Transform_transformPathWithFieldRef(path, field, "EPSG:4326");
```

**Reply by pyl-cool** [reply]
Hi Gonzalo，how can we have the shapes our field as a f2c.Cells called cells

**Reply by Gonzalo-Mier** [reply]
You can use the parser https://fields2cover.github.io/api/classf2c_1_1Parser.html#exhale-class-classf2c-1-1parser

**Reply by pyl-cool** [reply]
Thank you for your reply, can you provide a related tutorial, so that everyone better use?

**Reply by Gonzalo-Mier** [reply]
Unfortunately, lastly I don't have that much time. However, it's always a pleasure to accept PR :)

You can check the tests to see how to use it: https://github.com/Fields2Cover/Fields2Cover/blob/main/tests/python/utils/parser_test.py

**Reply by pyl-cool** [reply]
Sorry to bother you again, how can I add the starting point and the ending point and affect my final path planning result

**Reply by Gonzalo-Mier** [reply]
Unfortunately, this is not yet in the library.

**Comment by Botronics** [comment]
Hi @Gonzalo-Mier  for the conversion into gps path, all these steps has to be included in python tutotorial in  headlandgenerated part or simply in the launch file. If you can help it will be great help for me sir.
